climada/util/plot.py

```diff
@@ -290,6 +293,103 @@ def geo_im_from_array(array_sub, coord, var_name, title,
 
     return axes
 
+
+def geo_scatter_categorical(array_sub, geo_coord, var_name, title,
+                            cat_name = None,
+                            pop_name=True, buffer=BUFFER,
+                            extend='neither', proj=ccrs.PlateCarree(),
+                            shapes=True, axes=None, **kwargs):
+    """Map plots for categorical data defined in array(s) over input
+    coordinates. The categories must be a finite set of unique
+    values as can be identified by np.unique() (int, float, strings, ...).
+    
+    The categories are shared among all subplots, i.e. are obtained from 
+    np.unique(array_sub). 
+    Eg.:
+        array_sub = [[1, 2, 1.0, 2], [1, 2, 'a', 'a']] 
+        -> categories mapping is [[0, 2, 1, 2], [0, 2, 3, 3]]
+        
+    Same category: 1 and '1'
+    Different categories: 1 and 1.0
+    
+    
+    Parameters
+    ----------
+    array_sub : np.array(1d or 2d) or list(np.array)
+        Each array (in a row or in  the list) are categorical values at
+        each point in corresponding geo_coord that are ploted in one subplot.
+    geo_coord : 2d np.array)
+        (lat, lon) for each point in a row. 
+        The same grid is used for all subplots.
+    var_name : str or list(str)
+        label to be shown in the colorbar. If one provided, the same is 
+        used for all subplots. Otherwise provide as many as subplots in
+        array_sub.
+    title : str or list(str)
+        Subplot title. If one provided, the same is
+        used for all subplots. Otherwise provide as many as subplots in
+        array_sub.
+    cat_name : dict
+        Categories name for the colorbar labels. 
+        Keys are all the unique values in array_cub, values are their labels.
+    pop_name  : bool, optional
+        add names of the populated places. The default is True.
+    buffer  : float, optional
+        border to add to coordinates. The default is False.
+    extend : str, optional
+        extend border colorbar with arrows. The default is 'neither'.
+        options: [ 'neither' | 'both' | 'min' | 'max' ]
+    proj : ccrs
+        coordinate reference system used in coordinates
+    shapes : bool, optional
+        Overlay Earth's countries coastlines to matplotlib.pyplot axis.
+        The default is True.
+    kwargs : optional
+        arguments for hexbin matplotlib function.
```

Comment:
Maybe we can include all parameters that are already defined in `geo_scatter_from_array` in `kwargs` and also remove the redundant docstrings.

Reply:
I thought about that too. The annoying thing with this is that when one uses the helper overlay one sees only `kwargs` instead of the actual arguments.

Reply:
That doesn't justify the redundancy, in my opinion.

Reply:
Hmmm I will think about it :D

Reply:
Or you can use something like docrep (https://docrep.readthedocs.io/en/latest/)

Reply:
That looks great! Thanks

Reply:
The questions is here: add another dependency to CLIMADA or not?

Reply:
`docrep` is a really light-weight dependency. It only depends on a single, trivial package (`six`, no version requirement).

Reply:
ok, I will try it out for the current case

Reply:
I made a try with docrep. It works, but I am not yet convinced whether this is a good addition to the CLIMADA code base. It is a bit more difficult to read for non-expert python developers and not that much is gained.

Reply:
Thanks for checking it out!

The use of similar docstring replacements is a standard approach in many Python libraries. See https://github.com/scipy/scipy/blob/master/scipy/_lib/doccer.py and https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/docstring.py

I think, a non-expert climada user/developer should look into the compiled API reference instead of the source code.